### PR TITLE
fix(ape-ethereum): `.size` missing from `Block` for `newHeads` subs

### PR DESCRIPTION
**Pull request by fubuloubu**
### What I did

TEMP HACK: https://github.com/ApeWorX/ape/issues/1935

### How I did it

Realized that `eth_subscribe:newHeads` does not provide this info, so put a sentinel value in it's place

### How to verify it

Silverback apps work with Infura

### Checklist

- [x] All changes are completed
~~- [ ] New test cases have been added~~
~~- [ ] Documentation has been updated~~

### CODE REVIEW

**Review by antazoey — APPROVED** [review]

File: src/ape_ethereum/ecosystem.py [hunk 1]
```diff
@@ -551,6 +551,11 @@ def decode_block(self, data: Dict) -> BlockAPI:
         if "transactions" in data:
             data["num_transactions"] = len(data["transactions"])
 
+        if "size" not in data:
+            # NOTE: Due to an issue with `eth_subscribe:newHeads` on Infura
```
Comment (antazoey) [inline src/ape_ethereum/ecosystem.py:555]: Question: How can we re-design block API to handle a transaction-less state better.

Comment (fubuloubu) [reply src/ape_ethereum/ecosystem.py:555]: maybe combo of `_size: Optional[int] = Field(None, alias="size")` and a property `size(self) -> int` that either returns the cached value or triggers another call to fetch `.transactions` in order to compute the size from them

**Comment by antazoey** [comment]
@fubuloubu Any reason we've been holding off on merging this one?

**Comment by fubuloubu** [comment]
> @fubuloubu Any reason we've been holding off on merging this one?

Oh right! Lack of testing

Actually learning that IPC supports subscriptions is pretty cool, probably makes it easier to test

But for now, testing is probably not a deal breaker

**Review by antazoey — APPROVED** [review]

**Pull request merged**